public/app/features/alerting/unified/utils/rule-form.ts

```diff
@@ -587,6 +590,73 @@ export const panelToRuleFormValues = async (
   return formValues;
 };
 
+export const scenesPanelToRuleFormValues = async (
+  vizPanel: VizPanel,
+  variables: SceneVariable[],
+  scenesQueries: SceneQueryRunner,
+  scenesDashboard: DashboardScene
+): Promise<Partial<RuleFormValues> | undefined> => {
+  if (!vizPanel.state.key || !scenesDashboard.state.uid) {
+    return undefined;
+  }
+
+  const relativeTimeRange = rangeUtil.timeRangeToRelative(scenesDashboard.state.$timeRange?.state.value!); //TODO: how does $timeRange work, when is $timerange undefined? when is $timeRange.state.value undefined?
```

Comment:
What does this do?

Reply:
remember time range can exist on dashboard, panel or row level (technically), which is why we use sceneGraph